jetty-http/src/main/java/org/eclipse/jetty/http/HttpFields.java

```diff
@@ -738,6 +738,121 @@ public Mutable clear()
             return this;
         }
 
+        /** Ensure that specific HttpField exists when the field may not exist or may
+         * exist and be multi valued.
+         * @param field The header to ensure is contained.  The field is used
+         *              directly if possible so {@link PreEncodedHttpField}s can be
+         *              passed.  If the value needs to be merged with existing values,
+         *              then a new field is created.
+         */
+        public void ensure(HttpField field)
+        {
+            if (field.getValue().indexOf(',') < 0)
+            {
+                if (field.getHeader() != null)
+                    computeField(field.getHeader(), (h, l) -> computeEnsure(field, l));
+                else
+                    computeField(field.getName(), (h, l) -> computeEnsure(field, l));
+            }
+            else
+            {
+                if (field.getHeader() != null)
+                    computeField(field.getHeader(), (h, l) -> computeEnsure(field, field.getValues(), l));
+                else
+                    computeField(field.getName(), (h, l) -> computeEnsure(field, field.getValues(), l));
+            }
```

Comment:
I don't understand this logic.
The first if says "if the given field is single-valued"; but then `computeEnsure()` is passed the field, and there is no need to have two different `computeEnsure()`, the second with the values extracted from the field that is passed as first parameter :thinking:

Reply:
The implementation of the single field and multiple field versions are a bit different. Single field can already exist or not. Multi valued might partially exist.
I could simplify a `ensureField` a little, but only at the expense of complicating the `computeEnsure` methods.

Reply:
Note that I now have 100% test coverage of `ensureField` and both `computeEnsure`.